app/assets/javascripts/application.js

```diff
@@ -14,5 +14,41 @@
 //= require jquery_ujs
 //= require foundation
 //= require_tree .
-
+var votes;
 $(function(){ $(document).foundation(); });
+
+$(document).ready(function(){
+
+  $('.upvote-button').on('click', function(event){
+    event.preventDefault();
+    var button = $(this);
+    votes(button, 1);
+  });
+
+  $('.downvote-button').on('click', function(event){
+    event.preventDefault();
+    var button = $(this);
+    votes(button, -1);
+  });
+
+  $('.unvote-button').on('click', function(event){
+    event.preventDefault();
+    var button = $(this);
+    votes(button, 0);
+  });
+
+
+});
+
+votes = function(button, val) {
+  $.ajax({
+    method: 'POST',
+    url: button.parent().attr('action'),
+    dataType: 'json',
+    data: {vote: {value: val}}
+  })
+  .success(function(data){
+    var score = $(button.parent().parent().children('.score'));
+    score.text(data);
+  });
+};
```

Comment:
It's not a great idea to put javascript directly into this application.js file. I would extract out the functionality you need into a separate js file.